langchain4j-cloud-sql-postgres/src/main/java/dev/langchain4j/store/embedding/pgvector/PgVectorEmbeddingStore.java

```diff
@@ -0,0 +1,102 @@
+package dev.langchain4j.store.embedding.pgvector;
+
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.data.segment.TextSegment;
+import javax.sql.DataSource;
+import org.postgresql.ds.PGSimpleDataSource;
+import java.util.*;
+import dev.langchain4j.data.embedding.Embedding;
+
+
+import static dev.langchain4j.utils.HelperUtils.*;
+
+public class PgVectorEmbeddingStore implements EmbeddingStore<TextSegment> {
+
+    /**
+     * Constructor for PgVectorEmbeddingStore Class
+     * @param host                  the database host
+     * @param port                  the database port
+     * @param user                  the database user
+     * @param password              the database password
+     * @param database              the database name
+     */
+    protected PgVectorEmbeddingStore(
```

Comment:
Will we be creating a separate Engine class to manage the connection pooling? We should be using the Cloud SQL connector https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory?tab=readme-ov-file with arguments project id, region, cluster, instance, database, user, password, ip type, and iam account email once you can connect to a Cloud instance

Reply:
yes, we will use these arguments while connecting to cloud instance. Currently trying to resolve public IP/ private IP issue.